code401challenges/src/main/java/code401challenges/graph/Graph.java

```diff
@@ -1,20 +1,68 @@
 package code401challenges.graph;
 
-import java.util.Hashtable;
-import java.util.List;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.Set;
 
 public class Graph<T> {
-    public Hashtable<Vertex<T>, List<Edge<T>>> adjacencyList = new Hashtable<>();
-    private int size = 0;
+  private HashMap<T, ArrayList<Edge<T>>> vertices;
+  private int size = 0;
 
-    public Graph(Hashtable<Vertex<T>, List<Edge<T>>> adjacencyList) {
-        this.adjacencyList = adjacencyList;
+  public Graph(){this.vertices = new HashMap<>();}
+
+    public Graph(HashMap<T, ArrayList<Edge<T>>> vertices) {
+        this.vertices = vertices;
+    }
+
+    //Get nodes
+    public String getNodes() {
+        return vertices.keySet().toString();
+    }
+
+    public void setVertices(HashMap<T, ArrayList<Edge<T>>> vertices) {
```

Comment:
If you're going to give users the option to input an existing hash table into the graph then you also need to set the size to be the size of this inputted hash table